src/osd/OSDMap.cc

```diff
@@ -618,9 +618,8 @@ void OSDMap::Incremental::encode(bufferlist& bl, uint64_t features) const
     ENCODE_FINISH(bl); // osd-only data
   }
 
-  encode((uint32_t)0, bl); // dummy inc_crc
   crc_it = bl.end();
-  crc_it.advance(-4);
+  encode((uint32_t)0, bl); // dummy inc_crc
```

Comment:
I think we already have the terator-autofixing-machinery:

```cpp
  void buffer::list::iterator::copy_in(unsigned len, const char *src)
  {
    copy_in(len, src, true);
  }

  // copy data in
  void buffer::list::iterator::copy_in(unsigned len, const char *src, bool crc_reset)
  {
    // copy
    if (p == ls->end())
      seek(off);
    // ...
  }
```

`bl.end()` is used to capture the offset which `seek()` makes use of when it comes to `crc_it.copy_in()`.

Reply:
Wow that is gross but you are correct. Ooookaaaay then. :/
(Wanted to check this since you mentioned maybe dropping the last patch which eliminates this behavior.)